packages/contracts/contracts/discovery/Controller.sol

```diff
@@ -2,67 +2,82 @@
 pragma solidity =0.8.12;
 
 import {IController} from "./IController.sol";
-import {ProjectHelpers} from "../libraries/ProjectHelpers.sol";
+import {IProject} from "./IProject.sol";
+import {Project} from "./Project.sol";
+import {IBatch} from "./IBatch.sol";
+import {Batch} from "./Batch.sol";
 
-contract Controller is IController {
-    using ProjectHelpers for Project;
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
 
-    uint256 lastProjectId;
-    uint256 lastBatchId;
+contract Controller is IController, AccessControl {
+    mapping(address => bool) public projects;
+    mapping(address => address) public projectsToBatches;
 
-    mapping(uint256 => Project) public projects;
+    address public stakingContract;
 
-    mapping(uint256 => Batch) public batches;
+    event RegisterProject(address project);
```

Comment:
events should be in Past Tense, so `ProjectRegistered`

also, can we keep things organized here? (structs, then events, then state, like we have for the sale contracts)